code/modules/events/crystal_event.dm

```diff
@@ -509,9 +675,46 @@ This section is for the crystal monsters variations
 	icon_state = "crystal_boss"
 	icon_living = "crystal_boss"
 	icon_dead = "crystal_boss"
-	maxHealth = 150
-	health = 150
-	speed = 1
-	harm_intent_damage = 15
-	melee_damage_lower = 45
-	melee_damage_upper = 65
+	maxHealth = 300
+	health = 30
+	speed = 1.3
+	harm_intent_damage = 11
+	melee_damage_lower = 20
+	melee_damage_upper = 35
+	move_force = MOVE_FORCE_EXTREMELY_STRONG
+	move_resist = MOVE_FORCE_EXTREMELY_STRONG
+	pull_force = MOVE_FORCE_EXTREMELY_STRONG
+	environment_smash = ENVIRONMENT_SMASH_RWALLS
+	projectiletype = /obj/projectile/magic/aoe/lightning
+	projectilesound = 'sound/weapons/pierce.ogg'
+	ranged = 1
+	ranged_message = "throws"
+	ranged_cooldown_time = 25
+
+/mob/living/simple_animal/hostile/crystal_monster/boss/Bump(atom/clong)
+	. = ..()
+	if(isliving(clong))
+		var/mob/living/mob = clong
+		if(mob.stat >= HARD_CRIT)
+			mob.dust()
+			health += 35
+	if(isturf(clong))
+		var/turf/turf_bump = clong
+		turf_bump.Melt()
+	playsound(get_turf(src), 'sound/effects/supermatter.ogg', 35, TRUE)
+
+/mob/living/simple_animal/hostile/crystal_monster/boss/AttackingTarget()
+	. = ..()
+	if(isliving(target))
+		var/mob/living/mob = target
+		if(mob.stat >= HARD_CRIT)
+			mob.dust()
+			health += 35
+	else if(isturf(target))
+		var/turf/turf_bump = target
+		turf_bump.Melt()
+	else
+		qdel(target)
+	playsound(get_turf(src), 'sound/effects/supermatter.ogg', 35, TRUE)
```

Comment:
This is a pretty dangerous proc. Are you sure the things qdeleted here will be lawful ones?
Will this respect the `INDESTRUCTIBLE` flag of objects?
Can this delete abstract or visual objects?
How does the mob select its targets?

Reply:
mmm you're right, best not to have that else at the end with the qdel.. what about if i add on the living things that if they get hit and are not in hard crit they have a chance to loose a limb? would that be doable somehow? and if so what would be the best way to do that?

Reply:
`drop_limb()` is the proc you are looking for